STYLE.md

```diff
@@ -29,92 +34,140 @@ This document describes guidelines that people new to the codebase tend not to i
-  
-1. Use `static_cast`, not C++ functional-style casts or C casts.  Don't use `reinterpret_cast` where a `static_cast` would do -- use `static_cast` to cast to/from `void *`.
+  > Reason: `reinterpret_cast` can have different behavior than `static_cast`
+    (when multiple inheritance is involved), and it deserves a higher level of
+    scrutiny.  It's for when you're actually accessing raw data via two
+    different types.  Don't diminish its signal by using it for mere
+    void-pointer conversions to/from the same non-void type.
 
-  > Reason: `static_cast` is ugly and unmissable.  You can grep for it.  C-style casts can discard const qualifiers.  C++-style functional casts are usually somewhat bad -- you see them for signed/unsigned conversions and the like -- and making such things look scary is a good thing.
-  
-  > Reason: `reinterpret_cast` can have different behavior than `static_cast` (when multiple inheritance is involved), and it deserves a higher level of scrutiny.  It's for when you're actually accessing raw data via two different types.  Don't diminish its signal by using it for mere void-pointer conversions to/from the same non-void type.
+12. Avoid excessive dependency entanglements.  Prefer not to declare publicly
+used classes inside other classes.
 
-1. Avoid excessive dependency entanglements.  Prefer not to declare publicly used classes inside other classes.
+  > Reason: Declaring classes inside other classes means the class cannot be
+    forward declared, which makes refactoring header file dependencies much
+    more work.
 
-  > Reason: Declaring classes inside other classes means the class cannot be forward declared, which makes refactoring header file dependencies much more work.
-  
   > You could declare the classes inside a namespace instead.
 
-1. Run (from the src directory) `../scripts/check_style.sh` to see if your code has introduced any of a certain class of style errors.
+13. Run (from the src directory) `../scripts/check_style.sh` to see if your
+code has introduced any of a certain class of style errors.
 
-  > The include-what-you-use errors exist to make you suffer -- but keeping awareness of this makes efforts to untangle header file dependences a lot less labor-intensive.
+  > The include-what-you-use errors exist to make you suffer -- but keeping
+    awareness of this makes efforts to untangle header file dependences a lot
+    less labor-intensive.
 
   > Bogus errors can be suppressed with `// NOLINT(specific/category)`.
 
-  > This script has caught several bugs (mostly race conditions caused by not using reentrant "_r" versions of libc functions).
+  > This script has caught several bugs (mostly race conditions caused by not
+    using reentrant "_r" versions of libc functions).
```

Comment:
Would you consider using `"&#95;r"` instead of `"_r"` here since it causes syntax highlighting for Markdown to wig out (since `_` is an alternative form for italic syntax)?